apps/web/src/components/Settings/Dispatcher/ToggleDispatcher.tsx

```diff
@@ -22,14 +24,54 @@ interface Props {
 }
 
 const ToggleDispatcher: FC<Props> = ({ buttonSize = 'md' }) => {
+  const { data: idKitData } = useIsIdKitVerifiedQuery();
   const userSigNonce = useAppStore((state) => state.userSigNonce);
   const setUserSigNonce = useAppStore((state) => state.setUserSigNonce);
   const currentProfile = useAppStore((state) => state.currentProfile);
   const canUseRelay = currentProfile?.dispatcher?.canUseRelay;
 
+  const { setOpen: setIDKitOpen } = useIDKit({
+    autoClose: true,
+    enableTelemetry: true,
+    actionId: IDKIT_ACTION_ID,
+    handleVerify: async (result: ISuccessResult) => {
+      const response = await fetch(IDKIT_BRIDGE, {
```

Comment:
we always prefer using axios with tanstack query 🙇🏼, it would be great if we migrate to that 🚀

Reply:
Sure! I'm not super familiar with tanstack query, would love if you could provide an example on its usage with a POST endpoint somewhere in lenster I can look at

Reply:
We don't have any POST request in lenster so far in tanstack, but here is the simple example how to do

https://medium.com/analytics-vidhya/how-to-post-and-fetch-data-using-react-query-4c3280c0ef96

Reply:
Looking at the linked article, my understanding is that there's no way to get the response inline in the function, and you handle success/error states by adding `onSuccess`/`onError` callbacks on the widget.

This does unfortunately not work in this scenario, since `handleVerify` is ran by the IDKit widget, and we need to keep the state (or throw the error) inside of the function so IDKit can display the error to the user or know the validation has succeeded.

I can move `fetch` to `axios` however, if that fits the code conventions better, lmk